scripts/coverage_diff.py

```diff
@@ -107,9 +107,16 @@ def coverage_table(coverage, format='txt'):
     """
     columns = ["File Name", "Coverage (%)", "Delta Coverage"]
     separator = ['-'*len(cn) for cn in columns]
-    result = [columns, separator] + [[fn, coverage[fn]["child_cov"], coverage[fn]["delta_cov"]] for fn in coverage]
+    result = [columns, separator] + [[fn, coverage[fn]['child_cov'], coverage[fn]['delta_cov']] for fn in coverage]
     return txt_table(result) if format == 'txt' else markdown_table(result)
 
+def annotate_files(coverage):
+    """Return the exit code and exit message for given coverage
+    coverage: {filename: {parent_cov: float, child_cov: float, delta_cov: float, new_file: bool}}
+    output: None
+    """
+    [print(f"::notice file={fn},line=1,endLine=1,title=Coverage::{coverage[fn]['child_cov']}% ({coverage[fn]['delta_cov']}%)") for fn in coverage if abs(coverage[fn]['delta_cov']) > 1]
```

Comment:
Would it make sense to simply include the annotation in each `message` in `parse_coverage_status()`?

You could use different [message levels](https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#setting-a-notice-message) for each case, e.g.,
- `error` for ≥ 20% decrease
- `warning` for < 20% decrease
- `notice` for increase
- nothing if coverage is unchanged

Reply:
I have separated out the exit status into a separate function so that it could be reused in post message. Just to keep things separated and organized.